pkg/daemon/daemon.go

```
@@ -245,145 +348,80 @@ func (dn *Daemon) operatorConfigChangeHandler(old, new interface{}) {
-		err = dn.LoadedPlugins[GenericPlugin].Apply()
-		if err != nil {
-			glog.Errorf("nodeStateAddHandler(): generic_plugin fail to apply: %v", err)
-			dn.exitCh <- err
-			return
-		}
-	}
-
-	if reqReboot {
-		glog.Info("nodeStateAddHandler(): reboot node")
-		rebootNode()
-		return
-	}
-
-	// restart device plugin pod
-	if reqDrain {
-		if err := dn.restartDevicePluginPod(); err != nil {
-			glog.Errorf("nodeStateAddHandler(): fail to restart device plugin pod: %v", err)
-			dn.exitCh <- err
-			return
-		}
-	}
-
-	dn.completeDrain()
-	dn.refreshCh <- Message{
-		syncStatus:    "Succeeded",
-		lastSyncError: "",
-	}
-}
-
-func (dn *Daemon) nodeStateChangeHandler(old, new interface{}) {
+func (dn *Daemon) nodeStateSyncHandler(generation int64) error {
 	var err, lastErr error
-	newState := new.(*sriovnetworkv1.SriovNetworkNodeState)
-	oldState := old.(*sriovnetworkv1.SriovNetworkNodeState)
-	glog.V(2).Infof("nodeStateChangeHandler(): new generation is %d", newState.GetObjectMeta().GetGeneration())
+	glog.V(0).Infof("nodeStateSyncHandler(): new generation is %d", generation)
 	// Get the latest NodeState
 	var latestState *sriovnetworkv1.SriovNetworkNodeState
 	err = wait.PollImmediate(10*time.Second, 1*time.Minute, func() (bool, error) {
 		latestState, lastErr = dn.client.SriovnetworkV1().SriovNetworkNodeStates(namespace).Get(context.Background(), dn.name, metav1.GetOptions{})
 		if lastErr == nil {
 			return true, nil
 		}
-		glog.Warningf("nodeStateChangeHandler(): Failed to fetch node state %s (%v); retrying...", dn.name, lastErr)
+		glog.Warningf("nodeStateSyncHandler(): Failed to fetch node state %s (%v); retrying...", dn.name, lastErr)
 		return false, nil
 	})
 	if err != nil {
 		if err == wait.ErrWaitTimeout {
-			dn.exitCh <- goerr.Wrapf(lastErr, "nodeStateChangeHandler(): Timed out trying to fetch the latest node state %s", dn.name)
-			return
+			return fmt.Errorf("nodeStateSyncHandler(): Timed out trying to fetch the latest node state: %v", lastErr)
```

In theory, now we can just raise the error if the get fails, and the queue will handle the retry for us.

---

hmmm, it makes sense. I need to do some tests.

---

I removed some retries, it works. However, I cannot see the API timeout error in the latest 4.6 build anymore.